insonmnia/hub/miner_ctx.go

```diff
@@ -332,15 +331,17 @@ func (m *MinerCtx) registerRoutes(ID string, ports map[string]*pb.SocketAddr) []
 			continue
 		}
 
-		route, err := m.router.RegisterRoute(ID, binding.Network(), endpoint.Addr, uint16(endpoint.Port))
-		if err != nil {
-			log.G(m.ctx).Warn("failed to register route", zap.Error(err))
-			continue
+		for _, addr := range endpoint.Addrs {
+			route, err := m.router.RegisterRoute(ID, binding.Network(), addr, uint16(endpoint.Port))
```

Comment:
It's not ideal, because we actually can register all reals under the same VIP, but due to lack such API it's okay. I'll add it later.

Reply:
Yes, there's no suitable API yet. We'll change that later.